### PR TITLE
Update configurations and dependencies

### PR DESCRIPTION
**Pull request by mattnichols**
# Summary of Changes

* Remove deprecated PMD rules
* Add Hush plugin
* Upgrade detekt plugin
* Upgrade spotless plugin
* Fix Detekt configuration

## Downstream Consumer Impact

Need to test with major projects using plugin to ensure pipelines still work.

# How Has This Been Tested?

Publishing and testing with Path internal and external projects on Java 8 and Java 17

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by NateIorgMX** [comment]
I'll pull this into a few projects and test Java 8.

File: build.gradle [hunk 1]
```diff
@@ -21,12 +22,12 @@ repositories {
...
     implementation "ru.vyarus:gradle-quality-plugin:4.7.0"
-    implementation 'com.github.spotbugs.snom:spotbugs-gradle-plugin:5.0.9'
-    implementation ("com.diffplug.spotless:spotless-plugin-gradle:6.6.0") {
+    implementation 'com.github.spotbugs.snom:spotbugs-gradle-plugin:5.0.12'
+    implementation ("com.diffplug.spotless:spotless-plugin-gradle:6.11.0") {
         exclude group: "com.google.googlejavaformat", module: "google-java-format-parent"
     }
     implementation "com.google.googlejavaformat:google-java-format-parent:1.15.0"
-    implementation "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.20.0"
+    implementation "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.22.0-RC1"
```
Comment (NateIorgMX) [inline build.gradle:30]: I'm always paranoid about release candidates. Is there something we need from `1.22.0` specifically? I definitely trust your judgment, just curious.

File: build.gradle [hunk 2]
```diff
@@ -21,12 +22,12 @@ repositories {
 
 dependencies {
     implementation "ru.vyarus:gradle-quality-plugin:4.7.0"
-    implementation 'com.github.spotbugs.snom:spotbugs-gradle-plugin:5.0.9'
-    implementation ("com.diffplug.spotless:spotless-plugin-gradle:6.6.0") {
+    implementation 'com.github.spotbugs.snom:spotbugs-gradle-plugin:5.0.12'
+    implementation ("com.diffplug.spotless:spotless-plugin-gradle:6.11.0") {
```
Comment (stevecl5) [inline build.gradle:26]: Spotless plugin versions after 6.6.0 caused problems the last time we tried to update. I tried pulling in a snapshot version based on this branch, and I'm seeing the same issue. 

The problem appears when using Coppuccino in a project and updating dependencies in that project. Doing so adds a bunch of old dependencies to the lockfile. For example:

```
org.eclipse.jdt:org.eclipse.jdt.core:3.25.0=spotless-1911100560
org.eclipse.platform:org.eclipse.core.commands:3.9.800=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.core.contenttype:3.7.900=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.core.jobs:3.10.1100=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.core.resources:3.14.0=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.core.runtime:3.20.100=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.equinox.app:1.5.100=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.equinox.common:3.14.100=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.equinox.preferences:3.8.200=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.equinox.registry:3.10.100=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.jface.text:3.17.0=spotless-1452293708
org.eclipse.platform:org.eclipse.jface:3.22.100=spotless-1452293708
org.eclipse.platform:org.eclipse.osgi:3.16.200=spotless-1452293708,spotless-1911100560
org.eclipse.platform:org.eclipse.text:3.11.0=spotless-1452293708,spotless-1911100560
```

Dependencies like the ones shown above introduce 3 CVEs:

- CVE-2019-9843
- CVE-2020-8908
- CVE-2021-41033

This seems to be related to a [known issue](https://github.com/diffplug/spotless/issues/1215) which is still not resolved in version 6.11.0. I think we should stick with version 6.6.0 for now and add a note stating that the dependency should not be updated until the issue is resolved.

Comment (mattnichols) [reply build.gradle:26]: Ok. I'll move it back.

**Comment by mattnichols** [comment]
> I'll pull this into a few projects and test Java 8.

I pulled it into several projects (java 8 and 17). They all worked.

File: build.gradle (hunk 1 above)
Comment (mattnichols) [reply build.gradle:30]: If it was production code, I would agree. I'm on the fence with a linter. 🤷

**Comment by NateIorgMX** [comment]
> > I'll pull this into a few projects and test Java 8.
> 
> I pulled it into several projects (java 8 and 17). They all worked.

Ah, I misunderstood the `Downstream Consumer Impact` section.

**Review by stevecl5 — APPROVED** [review]

**Pull request merged**